src/routes/relay/pipes/sponsored-call-dto.validator.pipe.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  PipeTransform,
+  Injectable,
+  HttpException,
+  HttpStatus,
+  Inject,
+} from '@nestjs/common';
+
+import {
+  ISafeInfoService,
+  SafeInfoService,
+} from '../../../datasources/safe-info/safe-info.service.interface';
+import { SponsoredCallSchema } from '../entities/schema/sponsored-call.schema';
+import { SponsoredCallDto } from '../entities/sponsored-call.entity';
+
+@Injectable()
+export class SponsoredCallDtoValidatorPipe implements PipeTransform {
+  static schema = SponsoredCallSchema;
+
+  constructor(
+    @Inject(SafeInfoService) private safeInfoService: ISafeInfoService,
+  ) {}
+
+  async transform<T>(value: T): Promise<SponsoredCallDto> {
+    const result = await SponsoredCallDtoValidatorPipe.schema.safeParseAsync(
+      value,
+    );
+
+    if (!result.success) {
+      throw new HttpException(
+        'Validation failed',
+        HttpStatus.UNPROCESSABLE_ENTITY,
+        { cause: result.error },
```

Comment:
One concern that I have with this is that we might be exposing internal data/state to the clients.

The actual cause of the error we'd get it from the logs but the for the API contract with the clients I'd suggest a more general message.